Doc/library/unittest.mock-examples.rst

```diff
@@ -300,28 +298,29 @@ Mocking asynchronous context manager
 Since Python 3.8, ``AsyncMock`` and ``MagicMock`` have support to mock
 :ref:`async-context-managers` through ``__aenter__`` and ``__aexit__``.
 By default, ``__aenter__`` and ``__aexit__`` are ``AsyncMock`` instances that
-return an async function.
+return an async function. So you can do:
 
-    >>> class AsyncContextManager:
-    ...     async def __aenter__(self):
-    ...         return self
-    ...     async def __aexit__(self, exc_type, exc, tb):
-    ...         pass
-    ...
-    >>> mock_instance = MagicMock(AsyncContextManager())  # AsyncMock also works here
     >>> async def main():
-    ...     async with mock_instance as result:
-    ...         pass
-    ...
+    ...     async with MagicMock() as async_context_manager: # Replacing MagicMock with AsyncMock would also work
+    ...         print('Inside async context manager')
+    >>> asyncio.run(main())
+    Inside async context manager
+
+This would not work with Mock class, though, because it doesn't have the pre-assigned ``__aexit__`` and ``__aenter__`` methods.
+
+    >>> async def main():
+    ...     async with Mock() as context_manager:
+    ...         print('Inside async context manager')
     >>> asyncio.run(main())
-    >>> mock_instance.__aenter__.assert_awaited_once()
-    >>> mock_instance.__aexit__.assert_awaited_once()
```

Comment:
I think these 2 lines should be kept.

Reply:
The Mock doesn't have these two methods, that's why it will fail.